HBASE-29445 Add Option to Specify Custom Backup Location in PITR

vinayakphegde opened the pull request.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 32s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ HBASE-28957 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  8s |  |  HBASE-28957 passed  |
| +1 :green_heart: |  compile  |   0m 30s |  |  HBASE-28957 passed  |
| -0 :warning: |  checkstyle  |   0m  9s | [/buildtool-branch-checkstyle-hbase-backup.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-7153/1/artifact/yetus-general-check/output/buildtool-branch-checkstyle-hbase-backup.txt) |  The patch fails to run checkstyle in hbase-backup  |
| +1 :green_heart: |  spotbugs  |   0m 31s |  |  HBASE-28957 passed  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 48s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 29s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 29s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   0m  9s | [/buildtool-patch-checkstyle-hbase-backup.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-7153/1/artifact/yetus-general-check/output/buildtool-patch-checkstyle-hbase-backup.txt) |  The patch fails to run checkstyle in hbase-backup  |
| +1 :green_heart: |  spotbugs  |   0m 34s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  11m 11s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 41s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 10s |  |  The patch does not generate ASF License warnings.  |
|  |   |  28m 49s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-7153/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/7153 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 2e3ed54b66b4 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-28957 / 8ee4abbf139fd0486335c5f593c26dc9d1bc8942 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 84 (vs. ulimit of 30000) |
| modules | C: hbase-backup U: hbase-backup |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-7153/1/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 40s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ HBASE-28957 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 20s |  |  HBASE-28957 passed  |
| +1 :green_heart: |  compile  |   0m 26s |  |  HBASE-28957 passed  |
| +1 :green_heart: |  javadoc  |   0m 19s |  |  HBASE-28957 passed  |
| +1 :green_heart: |  shadedjars  |   8m 22s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 55s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 29s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 29s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 19s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   7m 43s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  25m 10s |  |  hbase-backup in the patch passed.  |
|  |   |  54m 10s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-7153/1/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/7153 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 480e87955328 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-28957 / 8ee4abbf139fd0486335c5f593c26dc9d1bc8942 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-7153/1/testReport/ |
| Max. process+thread count | 3564 (vs. ulimit of 30000) |
| modules | C: hbase-backup U: hbase-backup |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-7153/1/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

vinayakphegde commented:
> LGTM, please fix the unused import
> 
> nit: the title of this JIRA is to add the `--backup-path` in PITR, but this option was introduced in the previous commit, do we miss anything?
> 
> should the title of this JIRA be "Implement custom backup location in PITR "?

Sure, I will fix the import.
Yeah, I introduced the option in the previous commit only. but didn't implement the actual code. since the PR was too big. 
Regarding the jira header, I changed that now.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   2m 52s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ HBASE-28957 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 52s |  |  HBASE-28957 passed  |
| +1 :green_heart: |  compile  |   0m 18s |  |  HBASE-28957 passed  |
| +1 :green_heart: |  javadoc  |   0m 14s |  |  HBASE-28957 passed  |
| +1 :green_heart: |  shadedjars  |   5m 49s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 54s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 18s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 18s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 13s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 45s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |  23m 33s | [/patch-unit-hbase-backup.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-7153/2/artifact/yetus-jdk17-hadoop3-check/output/patch-unit-hbase-backup.txt) |  hbase-backup in the patch failed.  |
|  |   |  45m 55s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-7153/2/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/7153 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 26075ec1912d 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-28957 / 835cf42842df1bf61050ac0ff5f06fb86ef5a3da |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-7153/2/testReport/ |
| Max. process+thread count | 3621 (vs. ulimit of 30000) |
| modules | C: hbase-backup U: hbase-backup |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-7153/2/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

taklwu reviewed (approved):
Lgtm
